src/app/app.service.ts

```diff
@@ -39,4 +42,16 @@ export class AppService {
             SwaggerModule.setup('api-docs', app, document);
         }
     }
+
+    public async generateStatsReport(): Promise<ServiceReportDto> {
```

Comment:
It doesn't seem like this needs to be async for any reason.

Reply:
Not sure why you marked this resolved? It's still unnecessarily async.

Reply:
Its handful of one or other.  When we come back into this work, for version 2, we can decide if its really necessary then.  As architect, I do not have a problem with it being async for now.